### DISCUSSION TITLE
How to add index to a sub field?

### DISCUSSION THREAD

**Discussion by jonwhittlestone** [BeanieODM/beanie#495] [Question]
Hello.

I have the following object with example schema in mongodb.

```
{
  "_id" : ObjectId('123'),
  "buildingName": "House",
  
  "windows": [
    {
      "windowName": "paned-large",
      "location": "south side",
    },
    {
      "windowName": "paned-small",
      "location": "north side",
    }
  ]
}

```

How would I add an index to `windows.windowName` ?

I've tried the following and it didn't work - ie. the indexes weren't created when I inserted some documents.

**Tried**

```python
class Window(BaseModel):
    windowName: Indexed(  # type: ignore
        str,
        index_type=pymongo.TEXT  # type: ignore
    )
    location: str

class Building(Document):
    buildingName: str
    windows:  List[Window]

```

thanks

**Comment by roman-right** [comment]
For this you need to use the respective field in the Settings inner class: https://beanie-odm.dev/tutorial/indexes/#multi-field-indexes
